src/lib/stream/mod.rs

```diff
@@ -37,44 +37,37 @@ use self::{
 #[derive(Debug)]
 pub struct Stream {
     state: Arc<RwLock<Option<StreamState>>>,
+    pipeline_id: Arc<PeerId>,
+    video_and_stream_information: Arc<RwLock<VideoAndStreamInformation>>,
     terminated: Arc<RwLock<bool>>,
     watcher_handle: Option<tokio::task::JoinHandle<()>>,
 }
 
 #[derive(Debug)]
 pub struct StreamState {
-    pub pipeline_id: PeerId,
-    pub pipeline: Pipeline,
-    pub video_and_stream_information: VideoAndStreamInformation,
+    pub pipeline_id: Arc<PeerId>,
+    pub pipeline: Option<Pipeline>,
+    pub video_and_stream_information: Arc<RwLock<VideoAndStreamInformation>>,
     pub mavlink_camera: Option<MavlinkCamera>,
 }
 
 impl Stream {
     #[instrument(level = "debug")]
     pub async fn try_new(video_and_stream_information: &VideoAndStreamInformation) -> Result<Self> {
-        let pipeline_id = Manager::generate_uuid();
+        let pipeline_id = Arc::new(Manager::generate_uuid());
 
+        // Replace Redirect with Video
         let video_and_stream_information = {
             let mut video_and_stream_information = video_and_stream_information.clone();
 
             if matches!(
                 video_and_stream_information.video_source,
                 VideoSourceType::Redirect(_)
             ) {
-                let url = video_and_stream_information
-                    .stream_information
-                    .endpoints
-                    .first()
-                    .context("No URL found")?;
-
-                let Some(encode) = get_encode_from_stream_uri(&url).await else {
-                    return Err(anyhow!("No encode found for stream"));
-                };
-
                 video_and_stream_information
                     .stream_information
                     .configuration = CaptureConfiguration::Video(VideoCaptureConfiguration {
-                    encode,
+                    encode: VideoEncodeType::Unknown("?".to_string()),
```

Comment:
maybe "Redirect stream" or something like it would make it more clear than just an interrogation